builtin/credential/aws-ec2/path_login.go

```diff
@@ -265,60 +320,97 @@ func (b *backend) parseIdentityDocument(s logical.Storage, pkcs7B64 string) (*id
 // option is enabled on the registered role.
 func (b *backend) pathLoginUpdate(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	identityDocB64 := data.Get("identity").(string)
```

Comment:
I guess we can try decoding as base64 and fallback to regular string parsing otherwise. But I'm also happy just requiring base64 rather than have many ML posts about how to embed the JSON doc into curl properly.

It actually might not be a bad idea to suggest to users in the documentation that they use base64 with no wrapping (so if using `curl` piped to `base64` the `-w 0` option to `base64`) so that they don't need to worry about stripping or quoting those line endings.